trunner/harness/psh.py

```diff
@@ -52,14 +52,22 @@ class ShellHarness(IntermediateHarness):
 
     """
 
-    def __init__(self, dut: Dut, prompt: str, cmd: Optional[List[str]] = None, prompt_timeout: Optional[int] = -1):
+    def __init__(
+        self,
+        dut: Dut,
+        prompt: str,
+        cmd: Optional[List[str]] = None,
+        prompt_timeout: int = -1,
+        suppress_dmesg: bool = True,
+    ):
         super().__init__()
         self.dut = dut
         self.prompt = prompt
         self.cmd = " ".join(map(shlex.quote, cmd)) if cmd is not None else cmd
         self.prompt_timeout = prompt_timeout
+        self.suppress_dmesg = suppress_dmesg
 
-    def __call__(self, result: TestResult) -> TestResult:
+    def assert_self_prompt(self):
```

Comment:
IMHO `assert_prompt` name would be enough, not sure what `self` means in this context